app/src/main/java/com/randomappsinc/studentpicker/home/RenameListDialog.java

```diff
@@ -0,0 +1,57 @@
+package com.randomappsinc.studentpicker.home;
+
+import android.app.Dialog;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.FragmentManager;
+
+import com.afollestad.materialdialogs.DialogAction;
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.randomappsinc.studentpicker.R;
+import com.randomappsinc.studentpicker.utils.PreferencesManager;
+
+public class RenameListDialog extends DialogFragment {
+
+    public interface Listener {
+        void onRenameListConfirmed(int position, String newListName);
+    }
+
+    private FragmentManager fragmentManager;
+    private Listener listener;
+    private PreferencesManager preferencesManager;
+    private int position;
+
+
+    RenameListDialog(FragmentManager fragmentManager, Listener listener, PreferencesManager preferencesManager) {
+        this.fragmentManager = fragmentManager;
+        this.listener = listener;
+        this.preferencesManager = preferencesManager;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        return new MaterialDialog.Builder(getActivity())
+                .title(R.string.rename_list)
+                .input(getString(R.string.new_list_name), "", (dialog, input) -> {
```

Comment:
Can't pass an instance of Material Dialog to this class Because of the getString() need a context, we can pass it through the constructor of the RenameListDialog or we can keep using the Fragment manager

Reply:
We can pass a context in the constructor and this class can build the MaterialDialog itself. That abstracts away pretty much the entire implementation from the activity. You can find a similar-ish example to our use case here: https://github.com/Gear61/Random-Name-Picker/blob/master/app/src/main/java/com/randomappsinc/studentpicker/choosing/ChoicesDisplayDialog.java